### PR TITLE
Include app mode in bundle

### PR DESCRIPTION
**Pull request by jcheng5**
Possible values are "shiny", "rmd", and "staticRmd".
These values may need to change to align with connect.

### CODE REVIEW

File: R/bundle.R [hunk 1]
```diff
@@ -145,6 +184,7 @@ createAppManifest <- function(appDir, files, users) {
   # create the manifest
   manifest <- list()
   manifest$version <- 1
+  manifest$appmode <- appMode
```
Comment (kippandrew) [inline R/bundle.R]: FWIW shinyapps uses schema validation for manifests. I believe this will break deploying to shinyapps.

**Comment by jjallaire** [comment]
Two possible resolutions:

1. shinyapps does more liberal schema validation

2. we make appmode conditional on ! shinyapps

I'm open to either. Thoughts?

**Comment by jjallaire** [comment]
Here's a PR to handle it on the rsconnect side: https://github.com/rstudio/rsconnect/pull/13

File: R/bundle.R (hunk 1 above)
Comment (jcheng5) [inline R/bundle.R]: Nice catch @kippandrew

**Pull request merged**